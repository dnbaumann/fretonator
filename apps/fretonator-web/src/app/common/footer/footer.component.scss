@import '../../../styles/vars';
@import '../../../styles/functions';
@import '../../../styles/mixins';
@import '../../../styles/typography';

.footer {
  background-color: var(--black);
  box-sizing: border-box;
  padding: pxToRem($grid-unit * 6);
  border-top: pxToRem($grid-unit) solid var(--peach);
}

.footer__inner {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  max-width: 1280px;
  margin-left: auto;
  margin-right: auto;

  @media screen and (min-width: $screen-med) {
    flex-direction: row;
    align-items: flex-start;
    justify-content: space-between;
  }
}

.footer__leftColumn {
  justify-content: center;
  display: flex;
  flex-direction: column;
  align-items: center;

  @media screen and (min-width: $screen-med) {
    justify-content: flex-start;
    align-items: flex-start;
    flex-basis: 40%;
  }
}

.footer__rightColumn {
  justify-content: center;
  display: flex;
  flex-direction: column;
  align-items: center;

  @media screen and (min-width: $screen-med) {
    justify-content: flex-end;
    align-items: flex-end;
    flex-basis: 35%;
  }
}

.footer__copy {
  @include font_bodyCopy;
  font-size: pxToRem(15);
  line-height: pxToRem(22);
  color: var(--offwhite);
  display: flex;
  flex-direction: column;
  justify-content: center;
  text-align: center;
  margin-bottom: pxToRem($grid-unit * 2);

  @media screen and (min-width: $screen-med) {
    justify-content: flex-start;
    text-align: left;
  }
}

.footer__copy--right {
  @media screen and (min-width: $screen-med) {
    justify-content: flex-end;
    align-self: flex-end;
    text-align: right;
  }
}

.footer__copy--link {
  @include text_link();
  color: var(--peach);
  border-color: var(--peach);
}

.footer__sectionTitle {
  @include font_medium;
  font-size: pxToRem(18);
  line-height: pxToRem(28);
  margin-top: pxToRem($grid-unit);
  margin-bottom: pxToRem($grid-unit);
  color: var(--peach);
  font-weight: var(--font-weight-bold);
  display: flex;
  text-align: center;

  @media screen and (min-width: $screen-med) {
    text-align: left;
  }
}

.footer__list {
  list-style: none;
  text-align: center;
  padding: 0;
  margin-bottom: pxToRem($grid-unit * 4);

  @media screen and (min-width: $screen-med) {
    text-align: left;
  }
}

.footer__listItem {
  @include font_bodyCopy;
  font-size: pxToRem(15);
  line-height: pxToRem(26);
  color: var(--offwhite);
}

.footer__listItemLink {
  @include text_link();
  color: var(--yellow);
  border-color: var(--yellow);

  @include focus_accessible();
}

.footer__copyright {
  color: var(--offwhite);
  font-size: pxToRem(12);
  letter-spacing: var(--letter-spacing-display);
  font-weight: var(--font-weight-bold);
  text-transform: uppercase;
  display: block;
  margin-top: pxToRem($grid-unit * 6);
  text-align: center;
}

.footer__bigButton {
  @include hard_button_base();
  display: inline-flex;
  align-self: center;
  margin-bottom: pxToRem($grid-unit * 6);

  @media screen and (min-width: $screen-med) {
    align-self: flex-start;
  }
}

.footer__smallButton {
  @include chip_button_base();
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  background-color: var(--black);
  margin-top: pxToRem($grid-unit * 4);
  padding: pxToRem($grid-unit / 2) pxToRem($grid-unit * 2);
  color: var(--yellow);
  border-color: var(--yellow);

  &:focus {
    border-color: transparent;
  }
}

.footer__support__svg {
  height: pxToRem(20);
  width: pxToRem(20);
  margin-right: pxToRem($grid-unit / 2);
}

.footer__twitchButton {
  background-color: #9147ff;
  color: var(--white);
  padding: pxToRem($grid-unit * 2);
  font-family: var(--font-family-main);
  font-weight: var(--font-weight-bold);
  border-radius: pxToRem(6);
  text-decoration: none;
  margin-top: pxToRem($grid-unit * 4);
}
