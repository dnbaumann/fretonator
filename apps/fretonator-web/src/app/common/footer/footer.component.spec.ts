import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { FooterComponent } from './footer.component';
import { Component } from '@angular/core';
import { FooterModule } from './footer.module';
import { RouterTestingModule } from '@angular/router/testing';

describe('FooterComponent', () => {
  @Component({
    selector: 'app-footer-spec',
    template: `
      <app-footer></app-footer>
    `
  })
  class FooterComponentSpec {
  }

  let component: FooterComponentSpec;
  let fixture: ComponentFixture<FooterComponentSpec>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [FooterComponentSpec],
      imports: [FooterModule, RouterTestingModule]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(FooterComponentSpec);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
    expect(component).toMatchSnapshot();
  });
});
