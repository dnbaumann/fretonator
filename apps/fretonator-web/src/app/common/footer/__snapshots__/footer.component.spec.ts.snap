// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterComponent should create 1`] = `FooterComponentSpec {}`;
